mesonbuild/cargo/manifest.py

```diff
@@ -0,0 +1,229 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright © 2022-2023 Intel Corporation
+
+"""Type definitions for cargo manifest files."""
+
+from __future__ import annotations
+import typing as T
+
+from typing_extensions import Literal, TypedDict
+
+EDITION = Literal['2015', '2018', '2021']
+CRATE_TYPE = Literal['bin', 'lib', 'dylib', 'staticlib', 'cdylib', 'rlib', 'proc-macro']
+
+Package = TypedDict(
+    'Package',
```

Comment:
they have keys with `-` in them (ie, `rust-version`). That' can't be represented with the class form because Pytho will interpret the `-` as the "minus operator" not as part of the string name.

Reply:
Oh, wonderful...

What about FixedPackage, FixedDependency, FixedBuildTarget though, which seems to exist to fix that? :D

I do see on second look that you're using the class form as well, later down. I'm guessing these 3 are holdovers?

Reply:
Would you prefer to maximize the class form, or to maximize the non-class form?

Reply:
I think we should use the class form as much as we can.

Reply:
Oh, now I saw your edit... sad.

Reply:
I've updated with class form as much as possible. There's also some wierdness with inheritance and total=False apparently, since I tried to create some extra shared classes in the middle of stuff to save on things common between the fixed and raw varieties.

Reply:
I also folded the PEP 655 stuff into the other patches, it didn't seem like anyone objected and it really did belong in the other patches.

Reply:
I think this thread is solved, right?